### PR TITLE
Introduce abstract base classes

### PR DESCRIPTION
**Pull request by asvetlov**
Fix for #102

### CODE REVIEW

**Comment by kxepal** [comment]
I hope you'll remove master merge commit on PR merge?

File: multidict/_multidict.pyx [hunk 1]
```diff
@@ -238,7 +239,7 @@ cdef class MultiDictProxy(_Base):
         """Return a copy of itself."""
         return self._base_class(self)
 
-abc.Mapping.register(MultiDictProxy)
+MultiMapping.register(MultiDictProxy)
```
Comment (webknjaz) [inline multidict/_multidict.pyx]: IMHO using it as a class decorator would look cleaner

Comment (kxepal) [reply multidict/_multidict.pyx]: Is that possible in Cython?

Comment (webknjaz) [reply multidict/_multidict.pyx]: cython must support decorators

Comment (asvetlov) [reply multidict/_multidict.pyx]: What decorators are your talking about?
Cython doesn't support deriving from abc classes IIRC.
`ABC.register()` is a standard way for registering C Extension classes

**Comment by asvetlov** [comment]
@kxepal I'll squash the PR on merging anyway.

**Comment by kxepal** [comment]
cool!

File: multidict/_multidict.pyx (hunk 1 above)
Comment (webknjaz) [reply multidict/_multidict.pyx]: What about
```python
@ABC.register
class ...:
```
?

Comment (asvetlov) [reply multidict/_multidict.pyx]: Aaah. Ok.

Comment (asvetlov) [reply multidict/_multidict.pyx]: Cython cannot apply decorators to cdef classes :(

Comment (kxepal) [reply multidict/_multidict.pyx]: That's what I expected.

**Pull request merged**